index.html

```diff
@@ -418,6 +415,46 @@ <h3>Obtaining Consent</h3>
       groupId</code></dfn>). This conveniently handles the common case of wanting
       to route both input and output audio through a headset or speakerphone
       device.</p>
+      <p>On page load, run the following step:</p>
+      <ol>
+        <li>
+          <p>On the <a data-cite=
+          "!HTML/#concept-relevant-global">relevant global
+          object</a>, create an internal slot: <dfn>[[\explicitelyGrantedAudioOutputDevices]]</dfn>,
+              used to store devices that the user grants explicitely through <code>selectAudioOutput</code>,
+              initialized to an empty set.</p>
+        </li>
+      </ol>
+      <p>This specification specifies the <dfn data-cite="!GETUSERMEDIA#device-exposure-decision-non-camera-microphone">
+        exposure decision algorithm for devices other than camera and microphone</dfn>.
+        The algorithm runs as follows, with <var>device</var> as input:
+        <ol>
+          <li>
+            <p>Let <var>document</var> be the
+            <a data-cite="!HTML/webappapis.html#current-settings-object">
+            current settings object</a>'s
+            <a data-cite="!HTML/webappapis.html#responsible-document">
+            responsible document</a>.</p>
+          </li>
+          <li>
+            <p>If <var>document</var> is not <a data-cite="!HTML/iframe-embed-object.html#allowed-to-use">
+            allowed to use</a> the feature identified by <a data-dfn-link-for="">"speaker-selection"</a>,
+            return <code>false</code>.</p>
+          </li>
+          <li>
+            <p>If <var>device</var> is in <a>[[\explicitelyGrantedAudioOutputDevices]]</a>, return <code>true</code>.</p>
+          </li>
+          <li>
+            <p>If <var>device</var> <a data-cite="!GETUSERMEDIA#def-mediadeviceinfo-groupId">groupId</a>
+            is the same as the <a data-cite="!GETUSERMEDIA#def-mediadeviceinfo-groupId">groupId</a>
+            of a microphone device that is exposed by <a><code>enumerateDevices()</code></a>,
```

Comment:
enumerateDecvices() is calling this procedure which is referencing back to enumerateDevices(). Can we have an internal slot and make sure that we run these steps after the other devices have already been added to the list-in-progress?